src/utils/progress.ts

```diff
@@ -103,27 +103,42 @@ export class Progress {
   }
 
   getPrediction(): string {
-    return timestampToShortTime(this.getPredictionTimestamp());
+    const timestamp = Number(this.getPredictionTimestamp());
+    if (Number.isNaN(timestamp)) return '';
+    const progress = new IntlDateTime(timestamp).getRelativeNowFriendlyText('Progress');
+    return progress;
   }
 
   getPredictionText(): string {
-    const pre = this.getPrediction();
-    if (pre) return api.storage.lang(`prediction_Episode_${this.type}`, [pre]);
-    return '';
+    const timestamp = Number(this.getPredictionTimestamp());
+    if (Number.isNaN(timestamp)) return '';
+    const dt = new IntlDateTime(Number(timestamp));
+    const time = dt.getRelativeNowText('Progress');
+    if (!dt.isValidDate()) return '';
+    if (dt.isNow()) return api.storage.lang('bookmarksItem_now');
+    return api.storage.lang(`prediction_Episode_${this.type}`, [time]);
   }
 
   getLastTimestamp(): number {
+    if (!this.getProgressLastTimestamp()) return NaN;
     return this.getProgressLastTimestamp();
   }
 
-  getLast(ago = true): string {
-    return timestampToShortTime(this.getLastTimestamp(), ago);
+  getLast(): string {
+    const timestamp = Number(this.getLastTimestamp());
+    if (Number.isNaN(timestamp)) return '';
+    const last = new IntlDateTime(timestamp).getRelativeNowFriendlyText('Progress');
+    return last;
   }
 
   getLastText(): string {
-    const last = this.getLast(false);
-    if (last) return api.storage.lang(`prediction_Last_${this.type}`, [last]);
-    return '';
+    const timestamp = Number(this.getLastTimestamp());
+    if (Number.isNaN(timestamp)) return '';
+    const dt = new IntlDateTime(Number(timestamp));
+    const time = dt.getRelativeNowText('Progress');
```

Comment:
I think we could use the long version here. What do you think?

Reply:
Seem logical to me. That string is used only in progress pill hover, so no need to save space